workflows/cloudify_system_workflows/snapshots/snapshot_restore.py

```diff
@@ -117,14 +110,55 @@ def restore(self):
                 self._restore_influxdb()
                 self._restore_credentials(postgres)
                 self._restore_agents()
-                self._restore_deployment_envs(existing_dep_envs)
                 self._restore_amqp_vhosts_and_users()
+                self._restore_deployment_envs()
             if self._restore_certificates:
                 self._restore_certificate()
         finally:
             ctx.logger.debug('Removing temp dir: {0}'.format(self._tempdir))
             shutil.rmtree(self._tempdir)
 
+    def _restore_deployment_envs(self):
+        deps = {}
+        for tenant in self._client.tenants.list():
+            ctx.logger.info(tenant)
+            tenant_name = tenant['name']
+            ctx.logger.info(tenant_name)
+            ctx._context['tenant']['original_name'] = ctx.tenant_name
+            ctx._context['tenant']['name'] = tenant_name
```

Comment:
I'm adding some more comments to explain what's going on here.

A context manager is an option but doesn't gain us a huge amount here since it's only used in this one place now, and the only way it can avoid having the context reset is if it raises an exception, which will mean the end of the execution anyway. Given that, I didn't feel like a context manager was worth using for this.